internal/db/repos/instance.go

```diff
@@ -256,15 +257,16 @@ func (r *InstanceRepository) GetByJobID(ctx context.Context, ownerID, jobID uint
 	return instances, nil
 }
 
-// GetByJobIDOrdered retrieves all instances for a given job ID, ordered by creation date (oldest first)
+// GetByJobIDOrdered retrieves all instances for a given job ID, ordered by creation date (oldest first) that are not terminated
 func (r *InstanceRepository) GetByJobIDOrdered(ctx context.Context, ownerID, jobID uint) ([]models.Instance, error) {
 	if err := models.ValidateOwnerID(ownerID); err != nil {
 		return nil, fmt.Errorf("invalid owner_id: %w", err)
 	}
 
 	query := r.db.WithContext(ctx).
 		Unscoped().
-		Where(&models.Instance{JobID: jobID})
+		Where(&models.Instance{JobID: jobID}).
+		Where("status != ?", models.InstanceStatusTerminated)
```

Comment:
[non blocking] follow up here, we should just extend the opts to this method so that we can filter by status and not just this one status.